android-agent/build.gradle.kts

```diff
@@ -63,15 +63,15 @@ android {
 
 dependencies {
     implementation(project(":common"))
-    implementation(project(":instrumentation:activity"))
     implementation(project(":instrumentation:crash"))
+    implementation(project(":instrumentation:startup")) // todo remove during the initializationevents adaptation work.
     implementation(libs.androidx.core)
     implementation(libs.androidx.navigation.fragment)
     implementation(libs.androidx.lifecycle.process)
 
     api(platform(libs.opentelemetry.platform))
     api(libs.opentelemetry.api)
-    implementation(libs.opentelemetry.sdk)
+    api(libs.opentelemetry.sdk)
```

Comment:
Are we saying that by including the `opentelemetry-android` project, an app is transitively importing the entire `opentelemetry-java` API and SDK, and can reference all those methods and objects even if they didn't import those projects directly?

Reply:
Yes. In this case it was [needed](https://github.com/open-telemetry/opentelemetry-android/pull/446/files#diff-4ca93ba3ce26d5157044736a24cd06a5705998161d2933bf6fc1e0c64ae05cbfR8) in the `startup` module which references the [Clock](https://javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-common/latest/io/opentelemetry/sdk/common/Clock.html) interface, which is available in the sdk (now that I think about it, it seems like it's a good interface for the OTel api lib). I guess we can also just duplicate that line straight into the `startup` gradle file and other modules that might need to reference an sdk specific type, is just that so far I didn't see the need for it. I'm curious to learn your thoughts about it, because based on what you mentioned it seems like exposing the OTel sdk classes to consumers of the Android agent might be a bad idea?

Reply:
I think it depends on what is needed, if we are exposing `api` just because we need to access some internals from OTel Java, that seems wrong.
If we are exposing because users might use features that are already implemented in the OTel Java, that seems just fine.

Reply:
Yeah, basically what Manoel said. Exporting a dependency transitively isn't ideal because you're basically forcing your dependency directly to your consumer. It makes sense if it's a deliberate choice and is stated explicitly - depending on X means you are also directly importing Y, so users can accidentally take a dependency on classes in Y without knowing it, and if they decided to remove X later from their dependency change, they'll still be glued to Y.

But I can see that in our case, it makes sense - this project is just an extension of `opentelemetry-java`, so depending on us means you're explicitly depending on that as well. If that's the stance we want to take, we should do that, but also document it. And if we go down this route, we should probably export the semconv module too, so apps that use us won't need to directly depend on any of the core OTel modules explicitly if they depend on us. Doing that also ensures that there are no version conflicts, which is always good.

Reply:
Other modules can also depend on `libs.opentelemetry.sdk` if needed as long as it's not a circular dependency :D

Reply:
Gotcha, from a first glance I guess I look at it as Hanson mentioned, i.e. "this project is an extension of the java SDK so it's fine to expose it" kind of view... Though I'm open to discuss it more in detail, given that said view that I mentioned might also mean exposing other libs as Hanson mentioned, so for now I'll just add the SDK straight into the modules that need it and will set some time in the SIG to talk this more in detail.